### PR TITLE
Cut release v97.2.0

### PR DESCRIPTION
**Pull request by jeddai**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,21 @@
+# v97.2.0 (_2023-03-08_)
+
+[Full Changelog](https://github.com/mozilla/application-services/compare/v97.1.0...v97.2.0)
+
+## General
+
+### ⚠️ Breaking Changes ⚠️
+
+- Android: The JVM compatibility target is now version 11 ([#5401](https://github.com/mozilla/application-services/issues/5401))
```
Comment (tarikeshaq) [inline CHANGELOG.md:9]: Aaah if this is a breaking change, we should release 98.0.0, right?

Unless our consumers are already compatible?

Comment (tarikeshaq) [reply CHANGELOG.md:9]: If so, lets change this to a "What's changed", and a part of the changelog mention that it's technically a breaking change, but our consumers won't break

**Review by tarikeshaq — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (jeddai) [reply CHANGELOG.md:9]: Great point, I'll reach out and ask the android folks. Looks like that change was by @rvandermeulen, I'll reach out to him and double check #5401

Comment (rvandermeulen) [reply CHANGELOG.md:9]: This change was already made in the downstream Android projects.

Comment (jeddai) [reply CHANGELOG.md:8]: Alright, this has been updated!

**Review by tarikeshaq — APPROVED** [review]
:shipit:

**Pull request merged**